spec/index.md

```diff
@@ -1566,9 +1566,11 @@ There must be at most one call to `ic0.call_on_cleanup` between `ic0.call_new` a
 
 -   `ic0.call_with_best_effort_response : (timeout_seconds : i32) -> ()`
 
-    Allows the system to generate a synthetic reject response without waiting for the actual response. If this happens, the reject callback will have a reject code of either `SYS_TRANSIENT` or `SYS_UNKNOWN` (the reject callback may still be triggered with other reject codes if the call is rejected for other reasons, e.g., the callee traps). Whenever the code is `SYS_TRANSIENT`, the caller can be sure that the call never made it to the callee. If it is `SYS_UNKNOWN`, then the call **may or may not** have been processed by the callee; for example, it could be that processing the call did not complete in time, that delivering the reply to the calling canister took too long, or simply that the system found it necessary to shed load. In this case, if the calling canister needs to know whether the call was successful, it must find an out-of-band way of doing so. For example, if the callee provides idempotent function calls, the calling canister can simply retry the call.
+    Relaxes the response delivery guarantee to be best effort, asking the system to respond at the latest after `timeout_seconds` have elapsed. The system can then also respond with a `SYS_UNKNOWN` reject code. This reject code signifies that the call **may or may not** have been processed by the callee; even if the callee produces a response, it will not be delivered to the caller.
 
-    The `timeout_seconds` parameter specifies how long the system should wait for a response before producing a timeout reject. It is silently bounded by the `MAX_CALL_TIMEOUT` system constant; i.e., larger timeouts are treated as equivalent to `MAX_CALL_TIMEOUT` and do not cause an error. The reject callback may be executed (possibly significantly) later than the specified time, e.g., if the canister is under high load. Also note that the system may, at any time, reject a call for other reasons (e.g., under high load); the calling canister should not rely on the reject callback not being called before timeout expiration.
+     The `timeout_seconds` is silently bounded from above by the `MAX_CALL_TIMEOUT` system constant; i.e., larger timeouts are treated as equivalent to `MAX_CALL_TIMEOUT` and do not cause an error. The reject callback may be executed (possibly significantly) later than the specified time, e.g., if the canister is under high load. The reject callback can also executed before timeout expiration, and possibly with a `SYS_UNKNOWN` code if the system is under load.
```

Comment:
```suggestion
     The `timeout_seconds` is silently bounded from above by the `MAX_CALL_TIMEOUT` system constant; i.e., larger timeouts are treated as equivalent to `MAX_CALL_TIMEOUT` and do not cause an error. The reject callback may be executed (possibly significantly) later than the specified time, e.g., if the canister is under high load. The reject callback may also be executed before timeout expiration, with a `SYS_UNKNOWN` code, if the system is under load.
```

Or

```suggestion
     The `timeout_seconds` is silently bounded from above by the `MAX_CALL_TIMEOUT` system constant; i.e., larger timeouts are treated as equivalent to `MAX_CALL_TIMEOUT` and do not cause an error. The reject callback may be executed (possibly significantly) later than the specified time, e.g., if the canister is under high load. If the system is under load, it may also execute the reject callback with a `SYS_UNKNOWN` code before timeout expiration.
```